flutter-idea/src/io/flutter/run/daemon/DevToolsServerTask.java

```diff
@@ -0,0 +1,399 @@
+
+          // Keep this printout so developers can see DevTools startup output in idea.log.
+          System.out.println("DevTools startup: " + text);
+          tryParseStartupText(text);
+        }
+      });
+      process.startNotify();
+    }
+    catch (ExecutionException e) {
+      cancelWithError(e);
+    }
+  }
+
+  private void setUpWithDart(GeneralCommandLine command) {
+    try {
+      this.process = new MostlySilentColoredProcessHandler(command);
+      this.process.addProcessListener(new ProcessAdapter() {
+        @Override
+        public void onTextAvailable(@NotNull ProcessEvent event, @NotNull Key outputType) {
+          tryParseStartupText(event.getText().trim());
+        }
+      });
+      process.startNotify();
+
+      ProjectManager.getInstance().addProjectManagerListener(project, new ProjectManagerListener() {
+        @Override
+        public void projectClosing(@NotNull Project project) {
+          devToolsFutureRef.set(null);
+          process.destroyProcess();
+        }
+      });
+    }
+    catch (ExecutionException e) {
+      cancelWithError(e);
+    }
+  }
+
+  private void setUpWithDaemon() {
+    try {
+      final GeneralCommandLine command = chooseCommand(project);
+      if (command == null) {
+        cancelWithError("Unable to find daemon command for project");
+        return;
+      }
+      this.process = new MostlySilentColoredProcessHandler(command);
+      daemonApi = new DaemonApi(process);
+      daemonApi.listen(process, new DevToolsService.DevToolsServiceListener());
+      daemonApi.devToolsServe().thenAccept((DaemonApi.DevToolsAddress address) -> {
+        if (!project.isOpen()) {
+          // We should skip starting DevTools (and doing any UI work) if the project has been closed.
+          return;
+        }
+        if (address == null) {
+          cancelWithError("DevTools address was null");
+        }
+        else {
+          devToolsFutureRef.get().complete(new DevToolsInstance(address.host, address.port));
+        }
+      });
+    }
+    catch (ExecutionException e) {
+      cancelWithError(e);
+    }
+
+    ProjectManager.getInstance().addProjectManagerListener(project, new ProjectManagerListener() {
+      @Override
+      public void projectClosing(@NotNull Project project) {
+        devToolsFutureRef.set(null);
+
+        try {
+          daemonApi.daemonShutdown().get(5, TimeUnit.SECONDS);
+        }
+        catch (InterruptedException | java.util.concurrent.ExecutionException | TimeoutException e) {
+          LOG.error("DevTools daemon did not shut down normally: " + e);
+          if (!process.isProcessTerminated()) {
+            process.destroyProcess();
+          }
+        }
+      }
+    });
+  }
+
+  private CompletableFuture<DevToolsInstance> checkForDartPluginInitiatedDevToolsWithRetries(ProgressIndicator progressIndicator)
+    throws InterruptedException {
+    progressIndicator.setText2("Waiting for Dart Plugin");
```

Comment:
Nit -- perhaps "Waiting for DTD Server to load"

Reply:
Done, switched to "Waiting for server with DTD"

Reply:
Nice.